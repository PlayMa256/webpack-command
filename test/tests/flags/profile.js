const { apply, build, crcDist, test, validate } = require('../../util');

const fixture = 'profile/profile';
const opts = { fixture };

let config;

test('--profile', module, () => {
  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should build`, () =>
    build(config).then((result) => {
      expect(/factory:\d+ms building:\d+ms = \d+ms/.test(result)).toBe(true);
      expect(crcDist()).toMatchSnapshot();
    }));
});
