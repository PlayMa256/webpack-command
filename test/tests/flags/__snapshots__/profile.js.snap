// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --profile > should apply #0 1`] = `
Object {
  "devtool": "source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "profile": true,
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --profile > should build #0 1`] = `"f349b18e"`;
